homework/index.js

```diff
@@ -1,46 +1,165 @@
 'use strict';
 
 {
-  function fetchJSON(url, cb) {
+  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  // defined for taking data from url
+  function fetchJSON(url, callBackFunction) {
     const xhr = new XMLHttpRequest();
     xhr.open('GET', url);
     xhr.responseType = 'json';
     xhr.onload = () => {
       if (xhr.status >= 200 && xhr.status <= 299) {
-        cb(null, xhr.response);
+        callBackFunction(null, xhr.response);
       } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        callBackFunction(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
       }
     };
-    xhr.onerror = () => cb(new Error('Network request failed'));
+    xhr.onerror = () => callBackFunction(new Error('Network request failed'));
     xhr.send();
   }
-
+  // defined for new element creat&append
   function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
+    const child = document.createElement(name);
+    parent.appendChild(child);
     Object.keys(options).forEach(key => {
       const value = options[key];
       if (key === 'text') {
-        elem.textContent = value;
+        child.textContent = value;
       } else {
-        elem.setAttribute(key, value);
+        child.setAttribute(key, value);
       }
     });
-    return elem;
+    return child;
+  }
+  // defined for error handling
+  function renderError(error) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, {
+      text: error.message,
+      class: 'alert',
+    });
+  }
+  // for creating select element
+  function createOptionElements(repositories, select) {
+    createAndAppend('option', select, {
+      text: 'Select a repository',
+      disabled: 'disabled',
+    });
```

Comment:
Contrary to what we did in the lecture, the assignment states that you should display the details of the first repo at application start-up. So you should remove this placeholder 'Select a repository' option and have the name of the first repo in the select box at startup. Also, you need to add code elsewhere to ensure that the corresponding details of that repo are shown at startup.

Reply:
No 'select repository' anymore